packages/unified-latex-util-parse/tests/parse.test.ts

```diff
@@ -73,4 +85,98 @@ describe("unified-latex-util-parse", () => {
         trim(ast);
         expect(ast).toEqual(targetAst);
     });
+
+    it("nested math subscripts", () => {
+        let ast = strToNodes("{1_2}", {
+            mode: 'math',
+        });
+        expect(ast).toEqual([{
+            type: "group",
+            content: [
+                AstBuilder.s("1"),
+                AstBuilder.m("_", AstBuilder.args([
+                    AstBuilder.arg([AstBuilder.s("2")], {
+                        openMark: '{',
+                        closeMark: '}',
+                    }),
+                ]), { escapeToken: "" }),
+            ],
+        }]);
+
+        ast = strToNodes("$x_{y_{\\text{hello there $p_q_r$}}}$");
+        expect(ast).toEqual([{
+            type: "inlinemath",
+            content:[
+                AstBuilder.s('x'),
+                AstBuilder.m("_", AstBuilder.args([
+                    AstBuilder.arg([
+                        AstBuilder.s("y"),
+                        AstBuilder.m("_", AstBuilder.args([
+                            AstBuilder.arg([
+                                AstBuilder.m("text", AstBuilder.args([
+                                    AstBuilder.arg([
+                                        ...textToStringNodes("hello there "),
```

Comment:
This concerns me, because it means that we aren't parsing as text at this point, but instead as math mode.

If you do `$\text{\url{my_url}}$` does it reprint as `$\text{\url{my_{u}rl}}$`?

Reply:
Got the problem finally. Add separate processing of `\text` macro inside math context.